MediaGalleryUi/Model/UploadImage.php

```diff
@@ -72,6 +81,7 @@ public function execute(string $targetFolder, string $type = null): void
         }
         $absolutePath = $mediaDirectory->getAbsolutePath($targetFolder);
         $uploadResult = $this->imagesStorage->uploadFile($absolutePath, $type);
+        $this->generateRenditions->execute($uploadResult['path'] . '/' . $uploadResult['file']);
```

Comment:
To cover all the cases of image upload:
 - Please move the generating renditions to synchronization process. Considering that `SynchronizeFilesInterface` is used here, using the renditions generation will cover both upload from media gallery and `bin/magento media-gallery:sync`
 - Please add renditions generation call to `\Magento\MediaGalleryIntegration\Plugin\SaveImageInformation::afterSave`
 - Please add renditions generation call to `\Magento\MediaGalleryIntegration\Plugin\SaveBaseCategoryImageInformation::afterMoveFileFromTmp`